### PR TITLE
RavenDB-19998 - Sharding - Deleting orchestrator doesn't change orchestrator topology's replication factor

### PR DESCRIPTION
**Pull request by lastav5**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19998

### Additional description

`Update()` function in `OrchestratorTopology` class was missing update of `ReplicationFactor`

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Testing by Contributor

- Tests have been added that prove the fix is effective or that the feature works

### CODE REVIEW

File: src/Raven.Client/ServerWide/DatabaseTopology.cs [hunk 1]
```diff
@@ -152,6 +152,8 @@ public void Update(DatabaseTopology topology)
             Stamp = topology.Stamp;
             PriorityOrder = topology.PriorityOrder;
             NodesModifiedAt = topology.NodesModifiedAt;
+
+            ReplicationFactor = topology.ReplicationFactor;
```
Comment (lastav5) [inline src/Raven.Client/ServerWide/DatabaseTopology.cs:156]: `DynamicNodeDistribution` is also not updated here. Is it on purpose?

Comment (karmeli87) [reply src/Raven.Client/ServerWide/DatabaseTopology.cs:156]: I guess not :)

Comment (karmeli87) [reply src/Raven.Client/ServerWide/DatabaseTopology.cs:156]: Let's fix this (+test) here as well

**Review by karmeli87 — APPROVED** [review]

File: src/Raven.Client/ServerWide/DatabaseTopology.cs (hunk 1 above)
Comment (lastav5) [reply src/Raven.Client/ServerWide/DatabaseTopology.cs:156]: I don't think there is any path that updates the distribution through this command. The dynamic-node-distribution toggle endpoint is not for sharding and also doesn't use `UpdateTopologyCommand`

**Pull request merged**